view/frontend/templates/js/replacejs.phtml

```diff
@@ -1583,7 +1583,8 @@ $isLoadTrackJsDynamic = $block->isDisableTrackJsOnNonBoltPages() && $isLoadConne
                     if (expectCartRendering) {
                         expectCartRendering = false;
                         try {
-                            if ((totals.base_subtotal+totals.base_discount_amount) * 100 == BoltState.boltCart.total_amount.amount) {
+                            if ((totals.base_subtotal+totals.base_discount_amount) * 100 == BoltState.boltCart.total_amount.amount
+                                <?= /* @noEscape */ $block->getAdditionalQuoteTotalsConditions(); ?>) {
```

Comment:
I'd have this return an array of callbacks instead of the (easy to improperly format) JS expression. Each callback will receive the result of the previous call, the first one being the default. For example:
```suggestion
                            var additionalQuoteTotalConditionCallbacks = [<?= /* @noEscape */ implode($block->getAdditionalQuoteTotalsConditions(), ','); ?>];
                            var totalsMatch = (totals.base_subtotal + totals.base_discount_amount) * 100 == BoltState.boltCart.total_amount.amount;
                            for (const i in additionalQuoteTotalConditionCallbacks) {
                                totalsMatch = additionalQuoteTotalConditionCallbacks[i].call(null, totalsMatch);
                            }
                            if (totalsMatch) {
```